python/tvm/relax/transform/lazy_transform_params.py

```diff
@@ -166,11 +165,15 @@ def visit_var_binding_(self, binding: relax.VarBinding) -> None:
             var_before_setitem = self.builder_.emit(value)
             # rewrite set item
             new_var = self.builder_.emit(
-                relax.Call(
+                # TODO(@relax-team): This is wrong! This is not pure,
+                # but there is no other way to allow this inside a dataflow block.
+                # Properly speaking, this pass should require ToNonDataflow first,
+                # but the liveness analysis requires dataflow blocks. This should be refactored
+                relax.call_pure_packed(
                     relax.ExternFunc("set_item"),
-                    [index, var_before_setitem],
-                    None,
-                    [relax.ObjectStructInfo()],
+                    index,
+                    var_before_setitem,
+                    sinfo_args=(relax.ObjectStructInfo(),),
```

Comment:
This is a total hack to get tests to pass, see bug report #14829. I don't think we should allow this call to be inserted inside a dataflow block.